src/ophyd_async/core/device.py

```diff
@@ -141,41 +145,36 @@ async def __aenter__(self) -> "DeviceCollector":
 
     async def _on_exit(self) -> None:
         # Name and kick off connect for devices
-        tasks: Dict[asyncio.Task, str] = {}
+        tasks: Dict[str, asyncio.Task] = {}
```

Comment:
Suggest that rather than storing these as tasks, create a dict of name -> coros in the for loop below, then call `wait_for_connection` on it at the end. Then we can ditch _wait_for_tasks